amanuensis/blueprints/notification.py

```diff
@@ -32,15 +33,15 @@ def retrieve_notifications():
 
 
 @blueprint.route("/all", methods=["GET"])
-def retrive_all_notification_log():
+def retrive_all_notification_log_by_user():
     try:
         logged_user_id = current_user.id
     except AuthError:
         logger.warning(
             "Unable to load or find the user, check your token"
         )
     with flask.current_app.db.session as session:
-        notification_schema = NotificationSchema(many=True)
-        all_messages = get_notifications(session, user_id = logged_user_id)
-        return notification_schema.dump(all_messages)
-
+        notificationlog_schema = NotificationLogSchema(many=True)
+        all_notifications = get_notifications(session, user_id = logged_user_id)
+        all_notification_logs = get_notification_logs(session, ids=[n.notification_log_id for n in all_notifications] , expired=False)
```

Comment:
This solution is easy and readable and uses existing functions. However it makes 2 DB calls instead of one. We could write a single query to join Notification and NotificationLog and that would make only one call to the DB. These are likely going to be light queries so probably it doesn't matter overengineering it. 
@paulmurdoch19 what is your opinion on this vs building a specific join query?

Reply:
I think keeping it that way should be fine since the query shouldn't be that big, if it gets to slow we can always come back and change it